tpb.go

```diff
@@ -30,53 +33,65 @@ func New(endpoints ...string) *Client {
 	}
 }
 
-func (c *Client) fetch(ctx context.Context, path string) ([]*Torrent, error) {
-	var err error
-	for i := 0; i < c.MaxTries; i++ {
-		endpoint := c.endpoints.best()
-		if endpoint == nil {
-			return nil, ErrMissingEndpoint
-		}
-
-		timeoutCtx, cancel := context.WithTimeout(ctx, c.EndpointTimeout)
-		defer cancel()
-
-		var torrents []*Torrent
-		torrents, err = fetchTorrents(timeoutCtx, endpoint.baseURL+path)
-		if err == nil {
-			return torrents, nil
-		}
-		endpoint.lastFailure = time.Now()
-
-		// Stop if the error is a parsing error
-		if errors.As(err, &ParserError{}) {
-			break
-		}
-
-		// Stop if the global context is done
-		if ctx.Err() != nil {
-			err = ctx.Err()
-			break
-		}
-	}
-
-	return nil, err
-}
-
-// Search will search Torrents
+// Search will search torrents
 func (c *Client) Search(ctx context.Context, search string, opts *Options) ([]*Torrent, error) {
 	if opts == nil {
 		opts = &DefaultOptions
 	}
-	path := "/search/" + search + opts.String()
-	return c.fetch(ctx, path)
+	search = fmt.Sprintf("%s", search)
```

Comment:
search is already a string, is it necessary to go through `fmt.Sprintf` ?

Reply:
True!